modules/amp_dfp/includes/amp_rtc_config.admin.inc

```diff
@@ -0,0 +1,277 @@
+    }
+    
+    for ($j = 0; $j < $form_state['num_macros'][$i]; $j++) {
+      $form['rtc_config']['wrapper']['vendor'][$i]['macros'][$j]['macro'] = array(
+        '#type' => 'textfield',
+        '#value' => isset($tag->settings['rtc_config']['vendor'][$i]['macros'][$j]['macro']) ? $tag->settings['rtc_config']['vendor'][$i]['macros'][$j]['macro'] : '',
+        '#parents' => array('rtc_config', 'wrapper', 'vendor', $i, 'macros', $j, 'macro'),
+      );
+      $form['rtc_config']['wrapper']['vendor'][$i]['macros'][$j]['value'] = array(
+        '#type' => 'textfield',
+        '#value' => isset($tag->settings['rtc_config']['vendor'][$i]['macros'][$j]['value']) ? $tag->settings['rtc_config']['vendor'][$i]['macros'][$j]['value'] : '',
+        '#parents' => array('rtc_config', 'wrapper', 'vendor', $i, 'macros', $j, 'value'),
+      );
+    }
+
+    $form['rtc_config']['wrapper']['vendor'][$i]['more_macro'] = array(
+      '#type' => 'submit',
+      '#value' => t('Add vendor macro'),
+      '#submit' => array('_amp_rtc_macro_form_submit'),
+      '#parents' => array('rtc_config', 'wrapper', 'vendor', $i),
+      '#limit_validation_errors' => array(),
+      '#ajax' => array(
+        'callback' => '_amp_rtc_macro_form_elements_callback',
+        'wrapper' => 'rtc-macro-table-' . $i,
+        'effect' => 'fade',
+      ),
+    );
+  }
+
+  $form['rtc_config']['urls'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Urls'),
+    '#collapsible' => TRUE,
+    '#description' => t('RTC url configuration.'),
+  );
+
+  if (empty($form_state['num_urls'])) {
+    $form_state['num_urls'] = 1;
+  }
+
+  $form['rtc_config']['urls']['url_wrapper'] = array(
+    '#type' => 'markup',
+    '#prefix' => '<div id="rtc-url-wrapper">',
+    '#sufix' => '</div>',
+    '#theme' => 'rtc_url_settings'
+  );
+
+  if (isset($tag->settings['rtc_config']['urls']) && count($tag->settings['rtc_config']['urls']) > $form_state['num_urls']) {
+    $form_state['num_urls'] = count($tag->settings['rtc_config']['urls']);
+  }
+
+  for ($k = 0; $k < $form_state['num_urls']; $k++) {
+    $form['rtc_config']['urls']['url_wrapper'][$k]['rtc_url'] = array(
+      '#type' => 'textfield',
+      '#value' => isset($tag->settings['rtc_config']['urls'][$k]['rtc_url']) ? $tag->settings['rtc_config']['urls'][$k]['rtc_url'] : '',
+    );
+
+    $form['rtc_config']['urls']['url_wrapper'][$k]['rtc_error_url'] = array(
+      '#type' => 'textfield',
+      '#value' => isset($tag->settings['rtc_config']['urls'][$k]['rtc_error_url']) ? $tag->settings['rtc_config']['urls'][$k]['rtc_error_url'] : '',
+    );
+  }
+
+  $form['rtc_config']['wrapper_timeout'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Timeout configuration'),
+    '#collapsible' => TRUE,
+    '#description' => t('RTC timeout configuration.'),
+    '#parents' => array('rtc_config'),
+  );
+
+  $form['rtc_config']['wrapper_timeout']['timeoutMillis'] = array(
```

Comment:
There is a mix of camel case and underscores, we should use underscores in Drupal and CamelCase in clases. i.e. timeoutMillis -> timeout_millis